File: app/src/main/java/com/example/myapplication/network/dto/DogListDto.java
package com.example.myapplication.network.dto;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class DogListDto {

    @SerializedName("dogs")
    private List<DetailInfo> dogs = new ArrayList<>();


    public List<DetailInfo> getDogs() {
        return dogs;
    }

    public void setDogs(List<DetailInfo> dogs) {
        this.dogs = dogs;
    }

    public static class DetailInfo implements Parcelable {
        @SerializedName("uid")
        private String userId;
        @SerializedName("isAdopted")
        private boolean isAdopted;
        @SerializedName("name")
        private String name;
        @SerializedName("type")
        private String type;
        @SerializedName("age")
        private int age;
        @SerializedName("gender")
        private String gender;
        @SerializedName("thumbnailImg")
        private String thumbnailImg;
        @SerializedName("imgs")
        private List<Img> imgs = new ArrayList<>();

        public String getUserId() {
            return userId;
        }

        public void setUserId(String userId) {
            this.userId = userId;
        }

        public boolean isAdopted() {
            return isAdopted;
        }

        public void setAdopted(boolean adopted) {
            isAdopted = adopted;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getType() {
            return type;
        }

        public void setType(String type) {
            this.type = type;
        }

        public int getAge() {
            return age;
        }

        public void setAge(int age) {
            this.age = age;
        }

        public String getGender() {
            return gender;
        }

        public void setGender(String gender) {
            this.gender = gender;
        }

        public String getThumbnailImg() {
            return thumbnailImg;
        }

        public void setThumbnailImg(String thumbnailImg) {
            this.thumbnailImg = thumbnailImg;
        }

        public List<Img> getImgs() {
            return imgs;
        }

        public void setImgs(List<Img> imgs) {
            this.imgs = imgs;
        }


        public static class Img  implements Parcelable {
            @SerializedName("img")
            private String img;

            public String getImg() {
                return img;
            }

            public void setImg(String img) {
                this.img = img;
            }


            @Override
            public int describeContents() {
                return 0;
            }

            @Override
            public void writeToParcel(Parcel dest, int flags) {
                dest.writeString(this.img);
            }

            public Img() {
            }

            protected Img(Parcel in) {
                this.img = in.readString();
            }

            public static final Creator<Img> CREATOR = new Creator<Img>() {
                @Override
                public Img createFromParcel(Parcel source) {
                    return new Img(source);
                }

                @Override
                public Img[] newArray(int size) {
                    return new Img[size];
                }
            };
        }

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel dest, int flags) {
            dest.writeString(this.userId);
            dest.writeByte(this.isAdopted ? (byte) 1 : (byte) 0);
            dest.writeString(this.name);
            dest.writeString(this.type);
            dest.writeInt(this.age);
            dest.writeString(this.gender);
            dest.writeString(this.thumbnailImg);
            dest.writeList(this.imgs);
        }

        public DetailInfo() {
        }

        protected DetailInfo(Parcel in) {
            this.userId = in.readString();
            this.isAdopted = in.readByte() != 0;
            this.name = in.readString();
            this.type = in.readString();
            this.age = in.readInt();
            this.gender = in.readString();
            this.thumbnailImg = in.readString();
            this.imgs = new ArrayList<Img>();
            in.readList(this.imgs, Img.class.getClassLoader());
        }

        public static final Parcelable.Creator<DetailInfo> CREATOR = new Parcelable.Creator<DetailInfo>() {
            @Override
            public DetailInfo createFromParcel(Parcel source) {
                return new DetailInfo(source);
            }

            @Override
            public DetailInfo[] newArray(int size) {
                return new DetailInfo[size];
            }
        };
    }
}

File: app_kotlin/src/main/java/com/example/myapplication/network/dto/DogListDto.kt
package com.example.myapplication.network.dto

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class DogListDto {

    @SerializedName("dogs")
    var dogs: List<DetailInfo> = ArrayList()

    class DetailInfo : Parcelable {
        @SerializedName("uid")
        var userId: String? = null
        @SerializedName("isAdopted")
        var isAdopted: Boolean = false
        @SerializedName("name")
        var name: String? = null
        @SerializedName("type")
        var type: String? = null
        @SerializedName("age")
        var age: Int = 0
        @SerializedName("gender")
        var gender: String? = null
        @SerializedName("thumbnailImg")
        var thumbnailImg: String? = null
        @SerializedName("imgs")
        var imgs: List<Img> = ArrayList()


        class Img : Parcelable {
            @SerializedName("img")
            var img: String? = null


            override fun describeContents(): Int {
                return 0
            }

            override fun writeToParcel(dest: Parcel, flags: Int) {
                dest.writeString(this.img)
            }

            constructor() {}

            protected constructor(`in`: Parcel) {
                this.img = `in`.readString()
            }

            companion object {

                @SuppressLint("ParcelCreator")
                val CREATOR: Parcelable.Creator<Img> = object : Parcelable.Creator<Img> {
                    override fun createFromParcel(source: Parcel): Img {
                        return Img(source)
                    }

                    override fun newArray(size: Int): Array<Img?> {
                        return arrayOfNulls(size)
                    }
                }
            }
        }

        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(dest: Parcel, flags: Int) {
            dest.writeString(this.userId)
            dest.writeByte(if (this.isAdopted) 1.toByte() else 0.toByte())
            dest.writeString(this.name)
            dest.writeString(this.type)
            dest.writeInt(this.age)
            dest.writeString(this.gender)
            dest.writeString(this.thumbnailImg)
            dest.writeList(this.imgs)
        }

        constructor() {}

        protected constructor(`in`: Parcel) {
            this.userId = `in`.readString()
            this.isAdopted = `in`.readByte().toInt() != 0
            this.name = `in`.readString()
            this.type = `in`.readString()
            this.age = `in`.readInt()
            this.gender = `in`.readString()
            this.thumbnailImg = `in`.readString()
            this.imgs = ArrayList()
            `in`.readList(this.imgs, Img::class.java.classLoader)
        }

        companion object {

            @SuppressLint("ParcelCreator")
            val CREATOR: Parcelable.Creator<DetailInfo> = object : Parcelable.Creator<DetailInfo> {
                override fun createFromParcel(source: Parcel): DetailInfo {
                    return DetailInfo(source)
                }

                override fun newArray(size: Int): Array<DetailInfo?> {
                    return arrayOfNulls(size)
                }
            }
        }
    }
}
